compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -470,8 +470,10 @@ fn encode_ty<'tcx>(
         // (See https://rust-lang.github.io/unsafe-code-guidelines/layout/scalars.html#fixed-width-floating-point-types.)
         ty::Float(float_ty) => {
             typeid.push(match float_ty {
+                FloatTy::F16 => unimplemented!("f16_f128"),
                 FloatTy::F32 => 'f',
                 FloatTy::F64 => 'd',
+                FloatTy::F128 => unimplemented!("f16_f128"),
```

Comment:
@rcvalle: are there encodings for these types?

Reply:
There is some more discussion for these at https://github.com/rust-lang/rust/pull/114607#discussion_r1309679811 and https://github.com/rust-lang/rustc-demangle/pull/64 for this

Reply:
Okay

Reply:
I replied on the other issue, but will reply it here as well for easier reference. My recommendation is:

* For `f128`, if it's compatible with `__float128` regardless of the underlying representation (i.e., changes between IEEE 754R 128-bit floating point type or IBM extended double depending on the target), we should always encode it as 'g' as it will be how it will also be encoded by Clang. (FWIW, it seems Clang uses IEEE 754R 128-bit floating point type instead of IBM extended double for `__float128` by default for ppc64le since May 2023 and `__ibm128` also doesn't seem to have a distinct encoding specified in the Itanium C++ ABI mangling.)
* For `f16`, if it's compatible with IEEE 754R 128-bit floating point type (`__fp16` and `_Float16` in Clang), we should always encode it as 'Dh'; if it's compatible with bfloat16 (which is unlikely?), we should always encode it as 'DF16b'. 
* The 'Dd,  'De', and 'Df' encodings are for when the Rust compiler and Clang support decimal floating point types.

tl;dr.: We should always encode `f128` as 'g', and `f16` as 'Dh'.

Reply:
Thanks, I will update these in the upcoming PR to add intrinsics.

Total miss that they didn't make `__float128` mangle as `q` to be consistent with `f` and `d` :)

Reply:
Thank you! :)